replit_river/rpc.py

```diff
@@ -86,6 +87,11 @@ class ControlMessageHandshakeResponse(BaseModel):
     status: HandShakeStatus
 
 
+class PropagationContext(BaseModel):
+    traceparent: str
+    tracestate: str
```

Comment:
Based on below, will this mean that one of these two values will always be `""`? Would it be preferable to have them be `NotRequired[str]` instead?

Reply:
I'll go ahead and give them some default values. Both _should_ be set, but the way otel injects these at runtime is it sets each field separately so there will be a point in time where one is set and the other isn't.

tbh I think the protocol should just have an arbitrary `metadata` field on the transport message that is `dict[str, str]`, that decouples the river protocol from specific tracing injection formats. But that's for another day...

Reply:
Err last minute decided to just make these optional like all of our other similar fields, rather than defaulting to empty str